internal/exec/stages/files/filesystemEntries.go

```diff
@@ -248,6 +248,20 @@ func (s stage) mapEntriesToFilesystems(config types.Config) (map[types.Filesyste
 	return entryMap, nil
 }
 
+func (s *stage) mountAuto(dev, mnt string) error {
+	var err error
+	// try to mount all possible formats from config/v2_*/types/filesystem.go (without "swap")
+	for _, tryFormat := range []string{"ext4", "btrfs", "xfs", "vfat"} {
+		if err = s.Logger.LogOp(
+			func() error { return syscall.Mount(dev, mnt, tryFormat, 0, "") },
+			"mounting %q at %q (trying %q)", dev, mnt, tryFormat,
+		); err == nil {
+			return nil
+		}
+	}
+	return fmt.Errorf("failed to mount device %q at %q: %v", dev, mnt, err)
```

Comment:
with this you'll always return the latest error (by mounting using `vfat` format) which can be annoying for debugging purpose ?

Reply:
Thanks, I added the list of tried filesystems to the error message. All previous errors are also logged and not lost.